src/components/content/care-team/patient-careteam-filters.tsx

```diff
@@ -1,52 +1,41 @@
 import { CareTeamModel } from "@/fhir/models/careteam";
 import { CareTeamPractitionerModel } from "@/fhir/models/careteam-practitioner";
 import { ResourceMap } from "@/fhir/types";
-import { compact, maxBy } from "@/utils/nodash";
+import { isEqual, orderBy, uniqWith } from "@/utils/nodash";
 
 export const applyCareTeamFilters = (
   data: fhir4.CareTeam[],
   includedResources: ResourceMap
 ) => {
+  // 1. Make a list of CareTeamPractitionerModel and filter out any practioners that dont have an id.
+  // 2. Sort effectiveStartDate and practitionerId to ensure we remove older records.
+  // 3. UniqWith of the practioners to remove dupe records.
+
   const careTeamModels = data.map(
     (careteam) => new CareTeamModel(careteam, includedResources)
   );
 
-  const careTeamMemberMap = new Map<
-    string,
-    CareTeamPractitionerModel | undefined
-  >();
-
-  careTeamModels.forEach((careTeamObject) => {
-    careTeamObject.resource.participant?.forEach((participant) => {
-      const practitioner = careTeamObject.getPractitionerByID(
+  const careTeamPractitionerModels: CareTeamPractitionerModel[] = [];
+  careTeamModels.forEach((careTeam) =>
+    careTeam.resource.participant?.forEach((participant) => {
+      const practitioner = careTeam.getPractitionerByID(
         participant.member?.reference as string
       );
-      if (practitioner?.id) {
-        if (participant.member?.reference) {
-          const careTeamPractitioner = new CareTeamPractitionerModel(
-            careTeamObject,
-            practitioner
-          );
-          if (careTeamMemberMap.has(practitioner.id)) {
-            const latestPractitioner = careTeamMemberMap.get(practitioner.id);
-            if (latestPractitioner && careTeamPractitioner.effectiveStartDate) {
-              careTeamMemberMap.set(
-                practitioner.id,
-                maxBy(
-                  [careTeamPractitioner, latestPractitioner],
-                  "effectiveStartDate"
-                )
-              );
-            } else {
-              careTeamMemberMap.set(practitioner.id, careTeamPractitioner);
-            }
-          } else {
-            careTeamMemberMap.set(practitioner.id, careTeamPractitioner);
-          }
-        }
+      if (practitioner?.id && participant.member?.reference) {
+        careTeamPractitionerModels.push(
+          new CareTeamPractitionerModel(careTeam, practitioner)
+        );
       }
-    });
-  });
+    })
+  );
 
-  return compact([...careTeamMemberMap.values()]);
+  const sortedCareTeamPractitionerModels = orderBy(
+    careTeamPractitionerModels,
+    [(c) => c.effectiveStartDate, (c) => c.id],
+    ["desc", "desc"]
+  );
```

Comment:
These orderBy's feel easier to quickly read over when the property getters are just key names instead of functions
```suggestion
  const sortedCareTeamPractitionerModels = orderBy(
    careTeamPractitionerModels,
    ["effectiveStartDate", "id"],
    ["desc", "desc"]
  );
```

Reply:
are we going to get weird sorting if `effectiveStartDate`s have varied formats (ie like the date sort issues in the table). If so might need to instead make the array of property names/getter functions to normalize the date
```
const sortedCareTeamPractitionerModels = orderBy(
    careTeamPractitionerModels,
    [(c) => normalizeDateSomehow(c.effectiveStartDate), "id"],
    ["desc", "desc"]
  );
```